src/torchsearchsorted/utils.py

```diff
@@ -1,15 +1,25 @@
 import numpy as np
 
 
-def numpy_searchsorted(a: np.ndarray, v: np.ndarray, side='left'):
-    """Numpy version of searchsorted that works batch-wise on pytorch tensors
-    """
-    nrows_a = a.shape[0]
-    (nrows_v, ncols_v) = v.shape
-    nrows_out = max(nrows_a, nrows_v)
-    out = np.empty((nrows_out, ncols_v), dtype=np.long)
-    def sel(data, row):
-        return data[0] if data.shape[0] == 1 else data[row]
-    for row in range(nrows_out):
-        out[row] = np.searchsorted(sel(a, row), sel(v, row), side=side)
+def numpy_searchsorted(a: np.ndarray, v: np.ndarray,
+                       out: np.ndarray=None, side='left') -> np.ndarray:
+    """Batch-wise version of numpy's searchsorted"""
+    a = np.asarray(a)
+    v = np.asarray(v)
+    a, v = broadcast_arrays(a, v, axis=0)
+    if out is None:
+        out = np.empty(v.shape, dtype=np.long)
+    for i in range(v.shape[0]):
+        out[i] = np.searchsorted(a[i], v[i], side=side)
     return out
+
+
+def broadcast_arrays(*arrays, axis=0):
+    """Broadcast arrays along one axis, leaving other axes unchanged"""
+    if axis < 0:
+        raise ValueError(f"Negative axis not supported, got {axis}")
+    axis_size = max(a.shape[axis] for a in arrays)
+    return [
+        np.broadcast_to(a, (*a.shape[:axis], axis_size, *a.shape[axis + 1:]))
+        for a in arrays
+    ]
```

Comment:
I don't think it's so much elegant in this case. IMO you should revert to the original version that's more concise

Reply:
The thing is, in the original version, the `sel` function is called every time to select a row in the `a` and `v` matrices. This means that you get two function calls for every iteration of the for loop. On the other hand, we could make use of the native broadcasting functions of numpy/pytorch, that expand a dimension by setting the stride to 0 and need to be called only once.

Also, don't you like a separate, reusable and documented function better than a nested function?

Reply:
well, I don't think this would be reused a lot, since it's quite hidden in our module.
if you like this, you could actually replace the sel function by a one-liner of your np.broadcast call, that would maybe be the best option ? I agree that this is probably more elegant
did you try: is this the same speed as the `sel`, slower, or faster ?